server/routes/signup.js

```diff
@@ -0,0 +1,58 @@
+const {Chef, Customer} = require('../models/index')
+const express = require('express');
+const router = express.Router();
+const _ = require('lodash');
+const jwt = require('jsonwebtoken');
+const config = require('config');
+const Joi = require('joi');
+
+//console.log(Customer);
+//handle request for /signup
+//check if user exist and store new user info in db, return jwt token and user.
+router.post('/', async (req,res)=>{
+    const { error } = validate(req.body);
+    if (error) return res.status(400).send(error.details[0].message);
+
+    if (req.body.type === 'chef'){
+        usermodel = Chef;
+    }else if (req.body.type === "customer"){
+        usermodel = Customer;
+    }
+
+    return await storeUser(usermodel,req,res);
+});
+
+
+function validate(user){
```

Comment:
Can we build this validation into the User schema?

Reply:
Yes I could do that, but that may need more code written, using Joi will be much easier doing the validation.

Reply:
Yeah, we can always refactor later on if we have extra time